src/services/test-helpers/mock/mockApi.ts

```diff
@@ -517,6 +519,159 @@ const auctionsWinningsAt = () =>
+const assetTBalanceTwo = rococoRegistry.createType('u64', 20000000);
+
+const assetsInfo = () =>
+	Promise.resolve().then(() => {
+		const responseObj = {
+			owner: accountIdOne,
+			issue: accountIdTwo,
+			admin: accountIdTwo,
+			freezer: accountIdTwo,
+			supply: assetTBalanceOne,
+			deposit: balanceOfTwo,
+			minBalance: rococoRegistry.createType('u64', 10000),
+			isSufficient: trueBool,
+			accounts: rococoRegistry.createType('u32', 10),
+			sufficients: rococoRegistry.createType('u32', 15),
+			approvals: rococoRegistry.createType('u32', 20),
+			isFrozen: falseBool,
+		};
+
+		return rococoRegistry.createType('AssetDetails', responseObj);
+	});
+
+const assetsMetadata = () =>
+	Promise.resolve().then(() => {
+		const responseObj = {
+			deposit: balanceOfTwo,
+			name: rococoRegistry.createType('Bytes', 'statemint'),
+			symbol: rococoRegistry.createType('Bytes', 'DOT'),
+			decimals: rococoRegistry.createType('u8', 10),
+			isFrozen: falseBool,
+		};
+
+		return rococoRegistry.createType('AssetMetadata', responseObj);
+	});
+
+const assetBalanceObjOne = {
+	balance: assetTBalanceOne,
+	isFrozen: falseBool,
+	isSufficient: trueBool,
+};
+
+const assetBalanceObjTwo = {
+	balance: assetTBalanceTwo,
+	isFrozen: trueBool,
+	isSufficient: trueBool,
+};
+
+const assetBalanceObjThree = {
+	balance: assetTBalanceTwo,
+	isFrozen: falseBool,
+	isSufficient: falseBool,
+};
+
+const assetBalanceFactory = {
+	'10': rococoRegistry.createType('AssetBalance', assetBalanceObjOne),
+	'20': rococoRegistry.createType('AssetBalance', assetBalanceObjTwo),
+	'30': rococoRegistry.createType('AssetBalance', assetBalanceObjThree),
+};
+
+const assetAccountStorageKeyOne = statemintTypeFactory.storageKey(
+	10,
+	'AssetId',
+	statemintApiV1.query.assets.asset,
+	accountIdOne
+);
+
+const assetAccountStorageKeyTwo = statemintTypeFactory.storageKey(
+	20,
+	'AssetId',
+	statemintApiV1.query.assets.asset,
+	accountIdOne
+);
+
+const assetAccountStorageKeyThree = statemintTypeFactory.storageKey(
+	30,
+	'AssetId',
+	statemintApiV1.query.assets.account,
+	accountIdOne
+);
+
+const assetsAccountKeysAt = () =>
+	Promise.resolve().then(() => {
+		return [
+			assetAccountStorageKeyOne,
+			assetAccountStorageKeyTwo,
+			assetAccountStorageKeyThree,
+			assetAccountStorageKeyThree,
+			assetAccountStorageKeyThree,
+		];
+	});
+
+/**
+ * @param assetId options are 10, 20, 30
+ */
+interface AssetsAccount {
+	// eslint-disable-next-line  @typescript-eslint/no-explicit-any
+	[key: string]: any;
+}
+
+const assetsAccount: AssetsAccount = (
+	assetId: number | AssetId,
+	_address: string
+) => {
+	const id = typeof assetId === 'number' ? assetId : assetId.toNumber();
+
+	return Promise.resolve().then(() => {
+		switch (id) {
+			case 10:
+				return assetBalanceFactory[10];
+			case 20:
+				return assetBalanceFactory[20];
+			case 30:
+				return assetBalanceFactory[30];
+			default:
+				return;
+		}
+	});
```

Comment:
This doesn't look like it needs to be async at all?

```suggestion
	switch (id) {
		case 10:
			return Promise.resolve(assetBalanceFactory[10]);
		case 20:
			return Promise.resolve(assetBalanceFactory[20]);
		case 30:
			return Promise.resolve(assetBalanceFactory[30]);
		default:
			return Promise.resolve();
	}
```

Or just

```suggestion
	switch (id) {
		case 10:
			return assetBalanceFactory[10];
		case 20:
			return assetBalanceFactory[20];
		case 30:
			return assetBalanceFactory[30];
		default:
			return;
	}
```

Should be fine.

Reply:
Yup completely correct, it doesnt need to be async at all. Fixed